modules/react/popup/lib/hooks/useReturnFocus.tsx

```diff
@@ -116,7 +116,11 @@ export const useReturnFocus = createElemPropsHook(usePopupModel)(model => {
         return;
       }
 
-      if (requiresFocusChangeRef.current) {
+      if (
+        (document.activeElement !== document.body &&
+          !PopupStack.contains(model.state.stackRef.current, document.activeElement)) ||
+        requiresFocusChangeRef.current
+      ) {
```

Comment:
Update this logic to contain what is necessary without having to change `PopupStack`

Reply:
Updated!